packaging/CPackSourceInstall.cmake

```diff
@@ -0,0 +1,11 @@
+# CPACK_SOURCE_DIR is set to CMAKE_CURRENT_SOURCE_DIR in the root CMakeLists.txt
+# because CPack sets CMAKE_CURRENT_SOURCE_DIR to the path it is about to copy
+# the source into, but it has not yet copied the source when this script is run.
+set(GIT_REPO_ROOT "${CPACK_SOURCE_DIR}")
+include("${CPACK_SOURCE_DIR}/cmake/GitVersionInfo.cmake")
```

Comment:
This should read the CPack version instead of generate s new one.
Only this way the package version matches with the version in VersionInfo.cmake

Reply:
What? The VersionInfo.cmake must not exist except in the source packages.

Reply:
Here you have a redundant version.
One was created during the cmake configure stage and was written into the CPackConfigure file as various CPACK_xxx variables, the other is in the .git folder. 

If git changes in between the package and the bin version are different.

So you need to use here also one of the CPACK variables like the other packaging Scripts.

Reply:
Are you referring to the source DEB case? That remains to be updated.

Reply:
We could move the generation of the Debian package version to the CPackDebUploadPPA.cmake script if that's what you're suggesting.

Reply:
No, that would be against the normal cpack workflow.
CMake configured the CPack stage by writing all required info into the CPackConfigure and CPackConfigureSource files.
CPack reads the version number from there. This should be the source to write the version file, in the same way it writes the version into the file name.

Reply:
Would that create a window for error between the CMake configure step and cpack where the user might make a new commit or tag?

Reply:
I think it is more robust to always call Git at CPack time rather than rely on preconfigured variables.

Reply:
It is just the opositel calling git opens the wondow.

CPack calls Cmake if changes are applied to the source, so everything is fine.

Reply:
> CPack calls Cmake if changes are applied to the source, so everything is fine.

Are you sure? How do you know?

Reply:
Oh, right, I have just noticed that it does not work by default. You need to define the GitVersionInfo.cmake as install target excluded from ALL, and then:
set(CPACK_SOURCE_INSTALL_CMAKE_PROJECTS ${CMAKE_BINARY_DIR};mixx;GitVersionInfo;/)

Reply:
But it's not a target? It's only made with `configure_file`

Reply:
CMake is configured new if the result of configure_file is outdated. 
You must make sure that it gets that from git.

Reply:
Yes, cpack always gets the info from Git when making a source archive.